src/main/java/de/retest/web/RecheckSeleniumAdapter.java

```diff
@@ -61,18 +63,52 @@ public boolean canCheck( final Object toVerify ) {
 		logger.info( "Retrieving attributes for each element." );
 		final Set<String> cssAttributes = attributesProvider.getCssAttributes();
 		final JavascriptExecutor jsExecutor = (JavascriptExecutor) driver;
-		@SuppressWarnings( "unchecked" ) final Map<String, Map<String, Object>> result =
+		@SuppressWarnings( "unchecked" )
+		final Map<String, Map<String, Object>> rawAttributesMapping =
 				(Map<String, Map<String, Object>>) jsExecutor.executeScript( getQueryJS(), cssAttributes );
 
-		logger.info( "Checking website {} with {} elements.", driver.getCurrentUrl(), result.size() );
-		final RootElement lastChecked = convertToPeers( result, driver.getTitle(), shoot( driver ) );
+		logger.info( "Checking website {} with {} elements.", driver.getCurrentUrl(), rawAttributesMapping.size() );
+		final RootElement lastChecked = convertToPeers( rawAttributesMapping, driver.getTitle(), shoot( driver ) );
+
+		addChildrenFromFrames( driver, cssAttributes, lastChecked );
+
 		if ( driver instanceof UnbreakableDriver ) {
 			((UnbreakableDriver) driver).setLastActualState( lastChecked );
 		}
 
 		return Collections.singleton( lastChecked );
 	}
 
+	public void addChildrenFromFrames( final WebDriver driver, final Set<String> cssAttributes,
+			final RootElement lastChecked ) {
+		final JavascriptExecutor jsExecutor = (JavascriptExecutor) driver;
+		final List<Element> frames = de.retest.web.selenium.By.findElements( lastChecked.getContainedElements(),
+				element -> "iframe".equalsIgnoreCase( element.getIdentifyingAttributes().getType() )
+						|| "frame".equalsIgnoreCase( element.getIdentifyingAttributes().getType() ) );
+
+		logger.debug( "Found {} frames, getting HTML per frame.", frames.size() );
+		for ( final Element frame : frames ) {
+			final String frameId = frame.getIdentifyingAttributes().get( "id" );
+			if ( frameId == null ) {
+				// TODO Implement handling e.g. via name, XPaht, etc.
+				logger.error( "Cannot retrieve frame with id null from {}.", frame );
+				continue;
+			}
+			try {
+				logger.debug( "Switching to frame with ID {}.", frameId );
+				driver.switchTo().frame( frameId );
+				@SuppressWarnings( "unchecked" )
+				final Map<String, Map<String, Object>> frameAttributesMapping =
+						(Map<String, Map<String, Object>>) jsExecutor.executeScript( getQueryJS(), cssAttributes );
+				final RootElement frameContent = convertToPeers( frameAttributesMapping, "", null );
```

Comment:
Instead of having no title (empty string), we could reuse the frame ID like:

```suggestion
				final RootElement frameContent = convertToPeers( frameAttributesMapping, "frame-" + frameId, null );
```